### PR TITLE
Add try to ignore reading exception of some period

### PR DESCRIPTION
**Pull request by MiquelIR**
Ignore reading exceptions of some wrong periods to allow the whole file to be imported without the failing periods.
- [x] test that the meter.warnings have the expected result when an error reading file occurs

### CODE REVIEW

File: tests/test_tg.py [hunk 1]
```diff
@@ -87,6 +87,35 @@ def test_fix_circutor_Fhi_00000000000000000W(self):
...
+
+            :return:
+            """
+            res = []
+            for cnc in self.tg_xml.obj.Cnc:
+                concentrator = TG.Concentrator(cnc)
+                for meter in concentrator.get_meters():
+                    values = TG.Values(meter, 'S04', self.tg_xml.version)
+                    res = values.get()
+            assert len(res) == 14
```
Comment (ecarreras) [inline tests/test_tg.py]: Better use methods of [Unittests asserts](https://docs.python.org/2/library/unittest.html#test-cases)

**Comment by MiquelIR** [comment]
Currently notifying of the found exceptions with "print". Couldn't find a way to log the error. There is an option to raise a message: raise except_f1("blabla") But it makes the execution stop and we don't want that in this case.

File: tests/test_tg.py (hunk 1 above)
Comment (MiquelIR) [inline tests/test_tg.py]: Like this?
self.assertTrue(len(res) == 14)

File: switching/input/messages/TG.py [hunk 2]
```diff
@@ -145,35 +145,38 @@ def get_S04(self):
...
+                                     'r1': int(S04_values.get('R1%s' % value_value)),
+                                     'r2': int(S04_values.get('R2%s' % value_value)),
+                                     'r3': int(S04_values.get('R3%s' % value_value)),
+                                     'r4': int(S04_values.get('R4%s' % value_value)),
+                                     'value': value_value,
+                                     })
+                    ret_values.append(tmp_vals)
+            except Exception:
+                print "Unexpected error reading S04. Error:{}. On line {}" \
+                      "".format(sys.exc_info()[1], sys.exc_info()[2].tb_lineno)
```
Comment (tinogis) [inline switching/input/messages/TG.py]: We can find a way to get warnings after has been processed, like `get_warnings()`

File: tests/test_tg.py (hunk 1 above)
Comment (ecarreras) [inline tests/test_tg.py]: For me is more simple to do:
```python
self.assertEqual(len(res), 14)
```

**Comment by MiquelIR** [comment]
TG -> Values, now does the work of collecting the meter errors of ValueError kind and add them to the "warnings" property of the corresponding meter. But this is done only for S04 case. It should be implemented for the S04 S05 and S02 reports.

File: switching/input/messages/TG.py [hunk 3]
```diff
@@ -89,90 +115,117 @@ def get_timestamp(self, element, value):
...
+    def keep_meter_warnings(self):
+        error_string = "Unexpected ValueError reading {}. " \
+                       "Meter: {}. Error:{}. " \
+                       "On line {}".format(self.report_type,
+                                           self.meter.name,
+                                           sys.exc_info()[1],
+                                           sys.exc_info()[2].tb_lineno)
+        _new_warn = self.meter.warnings
+        _new_warn.append(error_string)
+        self.meter.warnings = _new_warn
```
Comment (ecarreras) [inline switching/input/messages/TG.py]: Why not simply:
```python
self.meter.warnings.append(error_string)
```

**Comment by ecarreras** [comment]
@MiquelIR test that the `meter.warnings` have the expected result when an error reading file occurs

**Pull request merged**